pkg/server/plugin/upstreamca/awssecret/awssecret_test.go

```diff
@@ -0,0 +1,182 @@
+	var (
+		err error
+	)
+	s.awsUpstreamCA, err = s.newAWSUpstreamCA()
+	s.Require().Nil(err)
+}
+
+func TestAWSSecret(t *testing.T) {
+	suite.Run(t, new(AWSSecretSuite))
+}
+
+func (as *AWSSecretSuite) TestConfigureNoGlobal() {
+	a := New(newFakeSecretsManagerClient)
+	req := new(spi.ConfigureRequest)
+	resp, err := a.Configure(nil, req)
+	as.Require().NotNil(err)
+	as.Require().Nil(resp)
+}
+func (as *AWSSecretSuite) TestGetSecret() {
+	svaluereq := secretsmanager.GetSecretValueInput{}
+	secretid := aws.String("cert")
+	svaluereq.SecretId = secretid
+	resp, err := as.client.GetSecretValueWithContext(ctx, &svaluereq)
+
+	as.Require().NotNil(resp)
+	as.Require().NoError(err)
+	as.Require().NotNil(aws.StringValue(resp.SecretString))
+	as.Require().True(strings.HasPrefix(aws.StringValue(resp.SecretString), "-----BEGIN CERTIFICATE-----"))
+	as.Require().NotNil(resp.ARN)
+}
+func (as *AWSSecretSuite) TestGetSecretFail() {
+	svaluereq := secretsmanager.GetSecretValueInput{}
+	secretid := aws.String("failure")
+	svaluereq.SecretId = secretid
+	resp, err := as.client.GetSecretValueWithContext(ctx, &svaluereq)
+
+	as.Require().NotNil(err)
+	as.Require().Nil(resp.SecretString)
+	as.Require().Nil(resp.ARN)
+}
+
+func (as *AWSSecretSuite) Test_SubmitValidCSR() {
+	const testDataDir = "_test_data/csr_valid"
+	validCsrFiles, err := ioutil.ReadDir(testDataDir)
+	as.Require().NoError(err)
+
+	for _, validCsrFile := range validCsrFiles {
+		csrPEM, err := ioutil.ReadFile(filepath.Join(testDataDir, validCsrFile.Name()))
+		as.Require().NoError(err)
+		block, rest := pem.Decode(csrPEM)
+		as.Require().Len(rest, 0)
+
+		resp, err := as.awsUpstreamCA.SubmitCSR(ctx, &upstreamca.SubmitCSRRequest{Csr: block.Bytes})
+		as.Require().NoError(err)
+		as.Require().NotNil(resp)
+	}
+}
+func (as *AWSSecretSuite) Test_SubmitInvalidCSR() {
+	const testDataDir = "_test_data/csr_invalid"
+	validCsrFiles, err := ioutil.ReadDir(testDataDir)
+	as.Require().NoError(err)
+
+	for _, validCsrFile := range validCsrFiles {
+		csrPEM, err := ioutil.ReadFile(filepath.Join(testDataDir, validCsrFile.Name()))
+		as.Require().NoError(err)
+		block, rest := pem.Decode(csrPEM)
+		as.Require().Len(rest, 0)
+
+		resp, err := as.awsUpstreamCA.SubmitCSR(ctx, &upstreamca.SubmitCSRRequest{Csr: block.Bytes})
+		as.Require().Error(err)
+		as.Require().Nil(resp)
+	}
+}
+
+func (as *AWSSecretSuite) TestFailConfiguration() {
+	config := AWSSecretConfiguration{
+		KeyFileARN:      "",
+		CertFileARN:     "",
+		TTL:             "1h",
+		AccessKeyID:     "keyid",
+		Region:          "us-west-2",
+		SecretAccessKey: "accesskey",
+	}
+
+	jsonConfig, _ := json.Marshal(config)
+	pluginConfig := &spi.ConfigureRequest{
+		Configuration: string(jsonConfig),
+		GlobalConfig:  &spi.ConfigureRequest_GlobalConfig{TrustDomain: "localhost"},
+	}
+
+	m := New(newFakeSecretsManagerClient)
+	_, err := m.Configure(ctx, pluginConfig)
+	as.Require().Error(err)
+}
+
+func (as *AWSSecretSuite) TestAWSSecret_Configure() {
```

Comment:
I'm not sure this test is adding value since `newAWSUpstreamCA()` does a Configure() as part of its setup (especially if you assert for failure on the Configure() call there)